s3_file_field/_multipart.py

```diff
@@ -61,8 +61,18 @@ class MultipartManager:
 
     part_size = mb(64)
 
-    def initialize_upload(self, object_key: str, file_size: int) -> PresignedTransfer:
-        upload_id = self._create_upload_id(object_key)
+    def initialize_upload(
+        self,
+        object_key: str,
+        file_size: int,
+        content_type: str | None = None,
+        content_disposition: str | None = None,
```

Comment:
I must be missing something here... when I try this locally in python 3.8 it doesn't appear to be a valid way to make a union type, though I see there's a [PEP for it](https://www.python.org/dev/peps/pep-0604/), but it is not accepted until 3.10. And yet CI is passing. What's going on?

Reply:
Nevermind, just noticed `from __future__ import annotations`. Carry on.

Reply:
I was confused by this as well. I tried to use this exact code in `_multipart_minio.py` and `_multipart_boto3.py` and it broke CI and raised errors in my editor. In this file it seems totally benign.

In lieu of any understanding I can just use `Union` instead.

Reply:
I defer to other maintainers of this repo as to whether they want this syntax to be allowed. Frankly, I really like it as a way of expressing union types, but it's clearly not yet familiar to mainstream python, and still requires an import side effect, so that might be a turnoff.

Reply:
I want us to have nice things, but I think we should move to this syntax as part of a separate PR and code style policy change.

The intent of `from __future__ import annotations` was just to enable delayed evaluation of annotations, which was necessary for...something here; if the code now works without that `__future__` change, I think we should cut it (until we deliberately need it again).